Code Review #3

hoguer opened the pull request.

richardedwardbrown3 committed:
adding cheese and wine and cheese tbd

kbonifacio committed:
Merge branch 'dev' into felix/seedDataCreation

Felix-Cadiz committed:
fixed merge conflicts

Felix-Cadiz committed:
fixed merged conflicts

kbonifacio committed:
fixed backend connection to client. Server is running with no errors.

richardedwardbrown3 committed:
created folder tests, api and db spec js. Api has a test for products and the /products route. Db is coming soon.

richardedwardbrown3 committed:
made tests folder and api and db spec js

Felix-Cadiz committed:
feat(navbar) Navbar showing

kbonifacio committed:
completed semantic fixes across the db files.

michellemalf committed:
Merge branch 'dev' into felix/navigation

michellemalf committed:
Merge pull request #33 from hoguer/felix/navigation

Felix/navigation

michellemalf committed:
Merge branch 'dev' into kriselda/backendConnect

michellemalf committed:
Merge pull request #32 from hoguer/kriselda/backendConnect

Kriselda/backend connect

Felix-Cadiz committed:
feat(front end navigation), Navbar is connecting to different routes and endpoints.

Felix-Cadiz committed:
feat(components/products)

richardedwardbrown3 committed:
Merge pull request #35 from hoguer/felix/allProducts

feat (components/allProducts)

richardedwardbrown3 committed:
testing to see the tests git commit

richardedwardbrown3 committed:
fairly close to finished with the tests our code will respond to thus far.

michellemalf committed:
Michelle/frontend (#37)

* all products component not yet working

* products: not complete, need help

Co-authored-by: Felix-Cadiz <88756231+Felix-Cadiz@users.noreply.github.com>

richardedwardbrown3 committed:
Met with Ryan, added to json test jest runInBand and test:watch as well as installed jsonWebToken.

richardedwardbrown3 committed:
Merge branch 'dev' into ricky/tests

richardedwardbrown3 committed:
Merge pull request #38 from hoguer/ricky/tests

Ricky/tests finally :)

Felix-Cadiz committed:
Felix/debug populate initial data (#42)

* Tables working, servers connected.

* feat(allProducts), all products are not showing

* feat(allProducts) Tables are populating correctly

Co-authored-by: Michelle M <89042600+michellemal@users.noreply.github.com>

kbonifacio committed:
Code Review #2 Fixes

kbonifacio committed:
Kriselda/products page (#47)

* added product api routes and router

* fixing merge conflicts to pull current dev branch

* code review 2 fixes #2

* feat(productPage): Displayed products and styled each card

Co-authored-by: Felix-Cadiz <88756231+Felix-Cadiz@users.noreply.github.com>

Felix-Cadiz committed:
feat (routes/users) - User seed data created (#49)

* Tables working, servers connected.

* feat(allProducts), all products are not showing

* feat(allProducts) Tables are populating correctly

* fixed merge issues

* fixed merge conflict

* feat(routes/users)

Co-authored-by: Kriselda Bonifacio <88959891+kbonifacio@users.noreply.github.com>

Felix-Cadiz committed:
Felix/fixed orders database adapters (#51)

* Tables working, servers connected.

* feat(allProducts), all products are not showing

* feat(allProducts) Tables are populating correctly

* fixed merge issues

* fixed merge conflict

* feat(routes/users)

* feat(ordersdb) fixed SQL.

Co-authored-by: Kriselda Bonifacio <88959891+kbonifacio@users.noreply.github.com>

kbonifacio committed:
Kriselda/nav bar cart (#52)

* added product api routes and router

* fixing merge conflicts to pull current dev branch

* code review 2 fixes #2

* feat(productPage): Displayed products and styled each card

michellemalf committed:
latest changes to login and register, not yet working (#53)

Co-authored-by: Felix-Cadiz <88756231+Felix-Cadiz@users.noreply.github.com>

richardedwardbrown3 committed:
This is the commit for my new branch with SingleProduct in it. (#50)

* This is the commit for my new branch with SingleProduct in it.

* couple changes

* Update init_db.js

* final commit tonight 1/31

* 5 fixes, env, ids, and

* now with all fixes and a pull from dev.

* Update db.spec.js

* Update db.spec.js

Co-authored-by: Felix-Cadiz <88756231+Felix-Cadiz@users.noreply.github.com>

michellemalf committed:
Michelle/loginregister (#58)

* latest changes to login and register, not yet working

* more changes t oregister, not yet working

* register form

Co-authored-by: Felix-Cadiz <88756231+Felix-Cadiz@users.noreply.github.com>

richardedwardbrown3 committed:
Fixing API and DB tests (#61)

kbonifacio committed:
Kriselda/single product client side (#55)

- Products showing with some CSS styling. 
- Users are able to access individual products and return to the all products page.

Felix-Cadiz committed:
Add orders routes and a middleware function to set the user on the req object (#59)

Commit:
Small Syntax Fixes (#63)

Commit:
fix merge conflicts

hoguer commented on db/orders.js:
```diff
@@ -0,0 +1,158 @@
+const client = require("./client");
+
+// Does not return order's products
```
let's get rid of these comments

hoguer commented on db/orders.js:
```diff
@@ -0,0 +1,158 @@
+            FROM orders AS o
+            INNER JOIN order_products AS op ON op."orderId" = o.id
+            INNER JOIN products AS p ON op."productId" = p.id;
+            WHERE "userID" = $1;
```
need to `return order`

hoguer commented on db/orders.js:
```diff
@@ -0,0 +1,158 @@
+};
+
+// Does not return order's products
+async function getOrdersByUser({id}) {
```
let's get rid of the curlies and define this with `getOrdersByUser(user_id)`

Felix-Cadiz replied:
Following the milestone guidelines.

hoguer commented on db/orders.js:
```diff
@@ -0,0 +1,158 @@
+};
+
+// Does not return order's products
+async function getOrdersByProduct({id}) {
```
let's get rid of the curlies and define this with `getOrdersByProduct(product_id)`

Felix-Cadiz replied:
Following the milestone guidelines.

hoguer commented on db/orders.js:
```diff
@@ -0,0 +1,158 @@
+};
+
+// Does not return order's products
+async function getCartByUser({id}) {
```
getCartByUser(user_id)

hoguer commented on db/orders.js:
```diff
@@ -0,0 +1,158 @@
+      };
+  };
+
+async function completeOrder({ id }) {
```
we could also just call updateOrder with id and status: "completed"

Felix-Cadiz replied:
Following the milestone guidelines.

hoguer commented on db/users.js:
```diff
@@ -7,10 +7,10 @@ async function createUser ({ firstname, lastname, email, imgURL, username, passw
         ON CONFLICT (username) DO NOTHING
-        RETURNING id, username
-        `, [username, hashedPassword]);
+        RETURNING *
```
we should either return explicit fields (`RETURNING id, username, ...`) or make sure we delete the password before we return.

hoguer commented on index.js:
```diff
@@ -13,6 +20,31 @@ server.use(express.json());
+      console.log('token: ', token); //
+
+      const userObj = jwt.verify(token, JWT_SECRET);
+      console.log('userObj: ', userObj);
```
remove these logs.

hoguer commented on routes/orders.js:
```diff
@@ -0,0 +1,69 @@
+ordersRouter.get("/cart", isLoggedIn, async (req, res, next) => {
+    const { id } = req.body
+    try {
+        const userCart = await getCartByUser({id});
```
if this function is updated to remove the unnecessary curlies, just need to remove them in all calls, too.

hoguer commented on routes/orders.js:
```diff
@@ -0,0 +1,69 @@
+    const { orderId } = req.params;
+    const { status, userId } = req.body;
+    try {
+        const deletedOrder = await deletedOrder({orderId, status, userId})
```
I think we need to write this deleteOrder function and import it.

Felix-Cadiz replied:
Commented out for now as this component was written before deleteOrder has been created before another collaborator.

hoguer commented on routes/products.js:
```diff
@@ -2,17 +2,20 @@ const productsRouter = require("express").Router();
 const { getProductById, getAllProducts, createProduct, getProductByName } = require("../db")
 //get all products   
 productsRouter.get("/", async (req, res, next) =>{
+    console.log("Got here!")
```
let's remove this log

hoguer commented on routes/products.js:
```diff
@@ -2,17 +2,20 @@ const productsRouter = require("express").Router();
 productsRouter.get("/:id", async (req, res, next) => {
+    const { id } = req.params;
     try {
+        const { id } = req.params;
```
let's get rid of one of these lines

hoguer commented on routes/users.js:
```diff
@@ -0,0 +1,125 @@
+const jwt = require("jsonwebtoken");
+require('dotenv').config();
+
+// NOT BE TO PUSHED INTO THE FINAL PRODUCT. 
```
let's remove this for now

hoguer commented on routes/users.js:
```diff
@@ -0,0 +1,125 @@
+
+    try {
+        const user = await getUserByUsername(username);
+        if (username && password) {
```
missing the bcrypt compare call here to compare the user's db password to what they gave us

hoguer commented on routes/users.js:
```diff
@@ -0,0 +1,125 @@
+    };
+});
+
+usersRouter.use((error, req, res, next) => {
```
could put this error handler into index.js instead of copying it to every individual routes file.

michellemalf committed:
Michelle/loginregister (#60)

Login and Register are both functional; however, the navigate feature after submission is not working.

michellemalf committed:
Michelle/loginregister (#66)

Login and Register Working!

michellemalf committed:
Michelle/account (#72)

Login, Register, and Account page is all functional.

kbonifacio committed:
Kriselda/update nav bar (#62)

Navbar is working and styled.

richardedwardbrown3 committed:
Ricky/products/productId test passes (#70)

Tests are functional.

kbonifacio committed:
Kriselda/products nav bar (#74)

Individual Wine and Cheese components have been updated.

Felix-Cadiz committed:
Felix/account styling  (#73)

account is functional, styling included

richardedwardbrown3 reviewed (commented):
Wow that's a lot! Go team!

Commit:
Backend piece of Stripe setup (#71)

Starter code for Stripe component.

kbonifacio committed:
Kriselda/home component (#77)

Home component built and styled- functional.

kbonifacio committed:
Kriselda/products page2 (#84)

Styling and search params functioning!

Felix-Cadiz committed:
Felix/reviewsTable (#78)

* Tables working, servers connected.

* feat(allProducts), all products are not showing

* feat(allProducts) Tables are populating correctly

* fixed merge issues

* fixed merge conflict

* feat(routes/users)

* feat(ordersdb) fixed SQL.

* feat(orderRoutes), Orders API Routes are complete

* feat(routes), Checkout Databases and API Routes are complete

* Updates to API Routes & DB Adapters

* Code Review 3

* fixed merge conflicts

* General styling to Login, Register and About Us.

* feat(reviews) Reviews db, seed data, and create function are functional.

Co-authored-by: Kriselda Bonifacio <88959891+kbonifacio@users.noreply.github.com>

michellemalf committed:
Michelle/adminbe (#85)

- Login and Register now working
- Styling done to the account page
- API Routes created

Commit:
Remove unused import

Felix-Cadiz committed:
Small Fixes for Code Review 3 (#69)
